### PR TITLE
move and refactor media button 

### PR DESCRIPTION
**Pull request by sarahkeh**
Dev plan:  
- [x] Create new Add Media button - match style of New Article button 
- [x] Add to Projects Component nav bar
- [x] Wrap button in QueryRenderer that uses Relay Modern environment (Newsletter code as reference) 
- [x] Update Mutations to use Relay Modern -> CreateProjectMediaMutation
- [x] Remove unused status updates
- [x] Update the success save to navigate to the new media item value 
- [x] Verify warning no longer appearing
- [x] Delete no longer used files
- [x] Test with video upload, image upload, url, and text

## Description

Moved the Add media button to the sidebar switched to Relay Modern for the createProjectMedia mutation.  After the item is created, the user is redirected to the created item's media page.   Since this requires the dbid of the item being created, there is a longer load time between creating the item and being redirected to the item page.  Before, the data would optimistically be added to the media list, but now the returned value from the save is required.  So, the form is replaced by a MediasLoading icon while the dbid is being created. Given how Relay Modern handles file uploads, the loading time appears to be the same regardless of the media type being created. 

References: TICKET-ID-1, TICKET-ID-2, …, TICKET-ID-N

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Tested text, url, video, and image file type upload.  Works both from the list view and the item page. 

## Things to pay attention to during code review

Was not able to delete the CreateProjectMediaMutation as it's still used in BlankMediaButton.  Wanted to convert the dialog to use the ConfirmProceedDialog but I needed the proceed button to be a form submission button.  

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [x] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [x] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

### CODE REVIEW

File: src/app/components/media/CreateMedia/CreateMediaButton.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        }
+      },
+    });
+  };
+
+  return (
+    <>
+      <ButtonMain
+        buttonProps={{
+          id: 'new-article-menu__open-button',
```
Comment (amoedoamorim) [inline src/app/components/media/CreateMedia/CreateMediaButton.js:85]: please change the id prefix to `create-media-button__`

Comment (sarahkeh) [reply src/app/components/media/CreateMedia/CreateMediaButton.js]: Updated!

File: src/app/components/media/CreateMedia/CreateMediaButton.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              <FormattedMessage defaultMessage="Cancel" description="Generic label for a button or link for a user to press when they wish to abort an in-progress operation" id="global.cancel" />
+            }
+            size="default"
+            theme="lightText"
+            variant="text"
+            onClick={() => setOpen(false)}
+          />
+          <ButtonMain
+            buttonProps={{
+              id: 'create-media-dialog__submit-button',
```
Comment (amoedoamorim) [inline src/app/components/media/CreateMedia/CreateMediaButton.js:132]: Found a couple ids more to be updated

**Review by brianfleming — COMMENTED** [review]
nice @sarahkeh I'm happy with the conversation we had on Jira resolving that one question. I will leave the approval to eng

I just wanted to shout out your workspace avatar for the record! 🦄 

Screenshot 2024-08-29 at 4 48 49 PM

File: src/app/components/media/CreateMedia/CreateMediaButton.js (hunk 2 above)
Comment (sarahkeh) [reply src/app/components/media/CreateMedia/CreateMediaButton.js]: Updated

**Review by amoedoamorim — APPROVED** [review]
Awesome stuff! :unicorn:

**Pull request merged**